### PR TITLE
Support square function.

### PR DESCRIPTION
**Pull request by fukatani**
This PR is for supporting `functions.square`. ( #1448 )
But I'm wondering how to avoid relative import `UnaryFunctionsTestBase`.

How about move it to `chainer.testing`? Or not using it?

### CODE REVIEW

**Comment by takagi** [comment]
> How about move it to chainer.testing? Or not using it?

I sent PR #1534 which provides a helper for testing such functions, you may use it after it will be merged. Before that, I suppose that duplicated base class would be acceptable.

**Comment by fukatani** [comment]
OK !
I'd wait for merger and reimplement test using helper class.
There are many duplicated codes in test, helper class seems useful!

**Comment by delta2323** [comment]
So it is better to check #1534 with higher priority, isn't it?

**Comment by takagi** [comment]
Yes, it would be better.

**Comment by delta2323** [comment]
got it.

File: chainer/functions/math/square.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def square(x):
+    """Elementwise square function.
+
+    .. math::
+       y_i = x_i ^ 2.
+
+    Args:
+        x (~chainer.Variable): Input variable.
```
Comment (delta2323) [inline chainer/functions/math/square.py]: Now, Chainer functions can accept numpy and cupy ndarrays as well (c.f. #1221). So could you change the acceptable types to that (c.f. #1687 )?

```
(chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray)
```

Comment (fukatani) [inline chainer/functions/math/square.py]: Sure! I fixed doc.

**Comment by delta2323** [comment]
Now #1534 is merged, we can use unary function testing tools and proceed this PR.

**Comment by takagi** [comment]
`unary_math_function_test` provides the default input data in range `[-1, 1]` and it is enough for `square` function that accepts real numbers including negative numbers. So you can remove `make_data` optional parameter to use the default data provider:

```Python
@testing.unary_math_function_test(F.Square())
class TestSquare(unittest.TestCase):
    pass
```

The custom `make_data` function is needed for `sqrt` function because it does not accept negative numbers.

**Comment by fukatani** [comment]
Sure. I fixed it.

**Comment by delta2323** [comment]
https://github.com/pfnet/chainer/issues/1740

**Comment by fukatani** [comment]
I got it! No problem :smiley:

**Comment by delta2323** [comment]
Now `unary_math_function_test` is renamed to `unary_math_function_unittest` by #1741. Could you fix decorator's name and merge the latest master branch?

**Comment by fukatani** [comment]
I fixed it! :smiley:

File: chainer/functions/math/square.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def square(x):
+    """Elementwise square function.
+
+    .. math::
+       y_i = x_i ^ 2.
+
+    Args:
+        x (~chainer.Variable): Input variable., :class:`numpy.ndarray` or
```
Comment (delta2323) [inline chainer/functions/math/square.py]: Description seems corrupted.

**Comment by fukatani** [comment]
Thanks!
I'm sorry for silly mistake. :cry:

**Comment by delta2323** [comment]
LGTM 👍

**Pull request merged**